Created a test version of the SearchMultipleBaseSpirits functionality

gojamesgo85 opened the pull request:
added a button under preferred spirits in the search function that allows multiple base spirits to be searched along with any other criteria. 

I don't think that I like it but I made it anyway. It's something that the user can toggle but it might require a small tutorial window for the user to understand.  

![Screenshot 2024-01-08 at 12 15 14 AM](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/95845784/5577a38f-6aa5-4289-a710-d770fbc00a3d)

It's also hard to see the benefit of the function without more cocktails to test with. 

We can trash it if you want but it's an option at least.

gojamesgo85 commented:
![Screenshot 2024-01-09 at 2 28 59 AM](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/95845784/06b50ad5-10f4-4c4b-9698-09e7ae10f808)

gojamesgo85 commented:
![Screenshot 2024-01-09 at 2 30 34 AM](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/95845784/345573f1-364c-4174-abe3-5a8eafeac88f)
I reworded the "Results for separate base spirits" function. I'm starting to kinda like it. But I'd like to add functionality to the results list view that makes the sections with which spirits actually matched instead of one long resulting list.

MatthewHunt84 reviewed (changes requested):
Good job! I'll start with what I don't like: I don't think the N/A components is necessary. We have everything already in flavors, and now things just show up twice:

![image](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/96077863/ccf283c2-dda7-4491-9a74-59d484c1cc30)
![image](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/96077863/cc4ea4c5-cf10-4c43-bbd1-057b36ea40b1)
My vote would be to revert that change, unless you feel strongly about what it's adding. I'll leave it up to you though. If we do keep it we have to fix things showing up twice.

### Now the good stuff:
The search functionality by spirit works well! Nice job. I think might be a bit confusing where it is - it's kinda hard to visualize what that toggle does before the results are displayed. Maybe we can add that on the results page instead so the user can toggle "match all" vs "match by spirit" 

Match all is how we used to have it, and in match by spirit cocktails show up in sections based on the spirit with the number of matches shown in the cocktail cell instead of the section header like this :
![image](https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/96077863/613dd9f7-405a-4223-abb8-33dde610d72e)

### Next steps:
More than anything, I'd say we're heading down a path where we split the search into a 'basic' user friendly version, and 'advanced' - would you agree?
Seems like the basic should be all ingredients, listed alphabetically like we have with the cocktails page, with the search bar at the bottom.
Then advanced can be all the tabs with the separate flavors/spirits etc.

If you agree, one change I'd like you to make before we go down that route is to look at how things are displayed and make sure it makes sense alphabetically. 
For instance, right now if you were to scroll down to **P** for prosecco it wouldn't show up because we have it under "_Chilled prosecco_" - prosecco would show under **C** instead.
Same with lemon juice, it would show under **F** instead of **L** because it's "_Fresh Squeezed Lemon Juice_" so lets revise that to make sure its alphabetical like:
Lemon juice (fresh squeezed)
Prosecco (chilled)
or whatever. 

Text me if you have questions, I'll work on the project tonight as well the swiftData stuff should be ready by tomorrow. 
You're killing it.

gojamesgo85 commented:
added NAIngredients to tags so that the SEARCHFORMULTIPLEBASESPIRITS works appropriately. 
I further refined the tags on all enums.

I also added an alert that won't allow SEARCHFORMULTIPLWBASESPIRITS to be activated unless base spirits are selected.

MatthewHunt84 commented on MetaCocktailsSwiftData/Model/Cocktails/ModernClassics/SunnySide.swift:
```diff
@@ -0,0 +1,36 @@
+                      textures: [.light],
+                      styles: [.sour, .shaken])
+
+let sunnySideBuild =  Build(instructions: [Instruction(step: 1, method: "Use Kold draft of Hoshizake cubes and the mint will the muddled while you're shaking."),
```
*or

MatthewHunt84 reviewed (approved):
Great work man, only one note:

<img width="400" alt="image" src="https://github.com/MatthewHunt84/MetaCocktailsSwiftData/assets/96077863/c1580bd3-5941-48e0-9f6f-71c6975781a3">

This just seems a bit confusing. Can we change this to something like 
"Sort by: Matches / Spirits" with some king of toggle?
